contracts/TrueReward/IEarnFinancialOpportunity.sol

```diff
@@ -1,4 +1,84 @@
 pragma solidity ^0.5.13;
 
-contract IEarnFinancialOpportunity {
+import "../TrueCurrencies/TrueRewardBackedToken.sol";
+import "./yTrueUSDInterface.sol";
+import "../TrueCurrencies/Proxy/OwnedUpgradeabilityProxy.sol";
+import "../TrueCurrencies/IFinancialOpportunity.sol";
+import "./IdGenerator.sol";
+
+contract IEarnFinancialOpportunity is IFinancialOpportunity {
+    using SafeMath for uint256;
+
+    yTrueUSDInterface public yToken;
+    TrueRewardBackedToken public token;
+    mapping (address => uint256) public yTokenBalance;
+
+    modifier onlyOwner() {
+        require(msg.sender == proxyOwner(), "only owner");
+        _;
+    }
+
+    modifier onlyToken() {
+        require(msg.sender == address(token), "only token");
+        _;
+    }
+
+    function configure(
+        yTrueUSDInterface _yToken,
+        TrueRewardBackedToken _token
+    ) public onlyOwner {
+        yToken = _yToken;
+        token = _token;
+    }
+
+    function proxyOwner() public view returns(address) {
+        return OwnedUpgradeabilityProxy(address(this)).proxyOwner();
+    }
+
+    function deposit(address _account, uint256 _amount) external returns(uint256) {
+        require(token.transferFrom(_account, address(this), _amount), "transfer from failed");
+        require(token.approve(address(yToken), _amount), "approve failed");
+        
+        uint256 balanceBefore = yToken.balanceOf(address(this));
+        yToken.deposit(_amount);
```

Comment:
it would surprise me if this did not return the number of shared minted

Reply:
it does not

Reply:
we can do the calculation on our end, but calling balance of is potentially safer